### PR TITLE
EdgeRequestContextFactory: Add service property

### PR DESCRIPTION
**Pull request by bsimpson63**
For service tokens generated by authenticate_service.

Will be used for r2, messenger, others to communicate with thing service for non-request tasks.

👓 @spladug @pacejackson

### CODE REVIEW

File: baseplate/core.py [hunk 1]
```diff
@@ -506,6 +528,11 @@ def session(self):
         """:py:class:`~baseplate.core.Session` object for the current context"""
         return Session(id=self._t_request.session.id)
 
+    @cached_property
+    def service(self):
+        """:py:class:`~baseplate.core.Service` object for the current context"""
+        return Service(self.authentication_token)
+
```
Comment (bsimpson63) [inline baseplate/core.py]: I'm planning on using this in thing service by first trying to get `request_context.user.id` and if that fails checking `request_context.service.name`. Does that sound reasonable?

File: baseplate/core.py [hunk 2]
```diff
@@ -358,6 +359,27 @@ def event_fields(self):
...
+        """Authenticated Service name.
+
+        :type: name string or None if context authentication is invalid
+        :raises: :py:class:`NoAuthenticationError` if there was no
+            authentication token, it was invalid, or the subject is not a
+            servce.
+
+        """
+        subject = self.authentication_token.subject
+        if not (subject and subject.startswith("service/")):
```
Comment (pacejackson) [inline baseplate/core.py]: Maybe set `"service/"` to a variable `service_prefix = "service/"`?

File: baseplate/core.py (hunk 1 above)
Comment (spladug) [reply baseplate/core.py]: Sounds OK to start out. If things get more complicated we might want to figure out how to turn the user context into application authorization more natively. I'd really love to hook into Pyramid's authorization stuff where possible.

**Comment by bsimpson63** [comment]
@spladug should I merge this now, or wait until right before a new release?

**Comment by spladug** [comment]
Now is fine!

**Pull request merged**